### ISSUE TITLE
Frontier API connection not operational, getting no confirmation email when trying to connect to API

### ISSUE THREAD

**Issue by Razzledazzk** [EDCD/EDDI#288]
## EDDI version in which issue found
2.4.6-b2

I thought I was on the 2.4.5, but havent been able to remove the beta version.

## Steps to reproduce

1. I installed EDDI 2.4.5, and while setting it up, noticed there was a newer version, 2.4.6-b2 and figured "oh I must have taken the wrong version"
I then uninstalled EDDI 2.4.5, as well as deleted the APP data under AppData\Roaming\EDDI.
And then installed the 2.4.6-b2

2. After learning that I was using a Beta version from my cousin, I proceded to correct it, once again uninstalling EDDI, as well as removing the APP data.
After that, I installed the 2.4.5, and was unable to connect to API.

3. I watched those youtube videos there was about setting up EDDI, as well as googled what I could, but couldnt find the issue.

I also tried installing the 2.3.0, as that was the only version which gave me an option to update, believing perhaps I had skipped a step somewhere, but that only updated to 2.3.1.

4. I uninstalled, double checked the version I downloaded, tried to disable all firewalls, virus scans etc., but keeps ending up without being able to connect to Frontier API, 

## Observed
5. It says I am on the 2.4.5 version in the program, but once I created the report, my system filled out the top, with the version 2.4.6-b2.

## Investigation
While trying it all out, I never once got an confirmation email with a code to use for the API, I was asked by my cousin if that was perhaps the missing step, but I checked my spam folders, and no mail was there.
Its status is under 
EDDI Status: Frontier API connection not operational
Frontier API: Unable to log in. This is usually a temporary issue with Frontier's service; please try again later.

I installed the EDMC app and with the same email and password, got the confirmation email as well as connect with the API, so I dont think it is in the email step.

I have used copy paste from notepad with the email and password, first on Frontier's homepage and then the same copy paste into the EDDI, and still gets no connection.

Yours sincerly
Martin Jensen

**Comment by Tkael** [comment]
Don't mind the template. For the moment, it's hardcoded to 2.4.6-b2. Your actual EDDI version is what you see in the app.

Did you use the same email and password that you used when you registered your game account with Frontier? It needs to match before Frontier will send a confirmation email.

**Comment by Razzledazzk** [comment]
Hello Tkael

I had hoped that the error was in the version, sadly it sounds like that isnt the solution.
I am certain I typed in the right mail and password.
I ended up typing them in notepad for visual confirmation, as well as using same copy paste on Frontier's homepage.

So yes to the email and password check. Sadly :-/

**Comment by Tkael** [comment]
Ok. Have you tried resetting your connection to the Frontier API? 
If yes, we can try to do a hard reset too, just navigate to %APPDATA%/EDDI and delete credentials.json to wipe any prior login details from EDDI's memory.

**Comment by Razzledazzk** [comment]
Tried both yes.
Error still persists.

I deleted the entire EDDI folder under %APPDATA% between each session, as well as trying to delete just credentials.json

Neither solved the issue.

a. tried the soft reset, both with and without a complete reinstall.

b. I tried deleting only the credentials.json in the EDDI folder, restart program, try log in. 
Then deleting credentials.json, uninstalling EDDI, reinstalling it and trying again.

c. I tried deleting EDDI folder, restart program, then Delete folder, reinstall program.

Are there any rules to the password in the software that could trigger this?
I tested out another API program under advice, and E:D Market connector seems to accept my email and password.

**Comment by Tkael** [comment]
No, EDDI does not try to impose any rules on your email or password. Would you please enable verbose logging, attempt to log in, and then click 'Report an issue' to generate an abbreviated log (it will appear on your desktop). Attach that log to this issue and we'll see what we can find out from the log.

**Comment by Razzledazzk** [comment]
[eddi_issue.zip](https://github.com/EDCD/EDDI/files/1537575/eddi_issue.zip)

Ok, just wanted to eliminate that possibility.
There you go, I cross my fingers, and thank you for the support :)

**Comment by Tkael** [comment]
I'm not seeing any errors, exceptions, or other irregularities in your log - EDDI sees the server and is trying to access data there, but without a valid login it's not able to get past (figuratively speaking) the doorstep. 
```
2017-12-07T02:17:40 EDDI:.ctor [I] EDDI access to the companion app is disabled 
2017-12-07T02:18:06 CompanionAppService:Instance [D] No companion API instance: creating one 
2017-12-07T02:18:06 EDDI:.ctor [I] EDDI access to the companion app is disabled 
2017-12-07T02:18:06 CompanionAppService:Profile [D] Entered 
2017-12-07T02:18:06 CompanionAppService:Profile [D] Service in incorrect state to provide profile (NEEDS_LOGIN) 
2017-12-07T02:18:06 CompanionAppService:Profile [D] Leaving 
2017-12-07T02:18:35 CompanionAppService:GetRequest [D] Entered 
2017-12-07T02:18:35 CompanionAppService:GetRequest [D] Leaving 
2017-12-07T02:22:16 CompanionAppService:Instance [D] No companion API instance: creating one 
2017-12-07T02:22:16 EDDI:.ctor [I] EDDI access to the companion app is disabled 
2017-12-07T02:22:16 CompanionAppService:Profile [D] Entered 
2017-12-07T02:22:16 CompanionAppService:Profile [D] Service in incorrect state to provide profile (NEEDS_LOGIN) 
2017-12-07T02:22:16 CompanionAppService:Profile [D] Leaving 
2017-12-07T02:22:43 CompanionAppService:GetRequest [D] Entered 
2017-12-07T02:22:43 CompanionAppService:GetRequest [D] Leaving 
2017-12-07T02:23:19 CompanionAppService:GetRequest [D] Entered 
2017-12-07T02:23:19 CompanionAppService:GetRequest [D] Leaving 
2017-12-07T02:23:29 CompanionAppService:GetRequest [D] Entered 
2017-12-07T02:23:29 CompanionAppService:GetRequest [D] Leaving 
2017-12-07T02:24:19 CompanionAppService:GetRequest [D] Entered 
2017-12-07T02:24:19 CompanionAppService:GetRequest [D] Leaving 
2017-12-07T02:25:12 CompanionAppService:GetRequest [D] Entered 
2017-12-07T02:25:12 CompanionAppService:GetRequest [D] Leaving 
```
Could it have been something simple, like having CAPS LOCK turned on while inputting your password?

**Comment by Razzledazzk** [comment]
I just resetted my password, and made it way to simple, to eliminate spelling errors as well as caps lock etc.

It didnt change.

The other API program, E:D Market Connector lost connection but found it again after typing in the new password.

Not sure what I got that is blocking EDDI.
As stated earlier, I never once recieved an email where I was surposed to get a code?`
Is it possible to set the program back to before that step?

**Comment by Tkael** [comment]
That's what deleting credentials.json was supposed to do. To confirm, you had exited completely out of both EDDI and VoiceAttack before trying that step?

**Comment by Razzledazzk** [comment]
Yes I was logged out of EDDI while changing password, as well as deleted the EDDI folder.
I do not have Voice attack installed - is this the missing link?
I am ready to try it out to see if it changes, but thought VoiceAttack was only if I wanted to talk to the system - I was going to use EDDI to make the ship experience more alive + getting GALNET info read to me.

The timeline was:
1. logged out of EDDI
2. Deleted APPDATA folder
3. Opened EDDI, on the EDDI page, the text "EDDI Status: Frontier API connection not operational" is active.
4. Went to Frontier API page, and with copy paste from notepad, copied the email as well as the easy new password in.
The text "You do not have a connecton to the Frontier API at this time. Please enter your Elite: Dangerous email adress and password below" is present.
5. Pressing NEXT, system is thinking, after a few seconds it greys out, and windows informs me the program isn't responding. 
6. 5 seconds later the system responds again, and the text "Unable to log in. This is usually a temporary issue with Frontier's service; please try again later"
7. Now did the soft reset in the bottom.
8. Step 4-5-6 repeats.

9 . Then I checked out the E:D Market Connector, and it showed me it had lost connection. I changed the password, and then it started to update again properly.

I just repeated the steps and it didn't change.

**Comment by Tkael** [comment]
VoiceAttack is not a requirement. I'm just asking to better understand context.

The freezing and temporary inability to connect is interesting. I'll check the relevant section of the code for clues as to what might be going on this evening.

**Comment by Tkael** [comment]
The message "Unable to log in.  This is usually a temporary issue with Frontier's service; please try again later" is a generic message that EDDI displays when it encounters an unhandled exception in this section of the code. So essentially it's telling me that there's an exception/error that EDDI didn't log.

I've built a fresh version 2.4.5 installer with some extra debug logging built around the Frontier API login (attached).
[EDDI-2.4.5.zip](https://github.com/EDCD/EDDI/files/1541216/EDDI-2.4.5.zip)

Please install this version, then repeat steps 1 - 8 above with verbose logging enabled and attach a new log. Thanks!

**Comment by Razzledazzk** [comment]
Just tried it as instructed, step 1-8.

Did a clean reinstall of the program, without either of the EDDI folders (APPDATA and also deleted the install folder under programs)

[eddi_issue.zip](https://github.com/EDCD/EDDI/files/1542266/eddi_issue.zip)

**Comment by Tkael** [comment]
Exception captured.
`2017-12-08T10:14:43 MainWindow.xaml:companionAppNextClicked [D] Unable to login. Exception: System.Net.WebException: The operation has timed out`

Since I've been able to log out and log back in to the Frontier API several times in the last few days, and I haven't had any other reports with people unable to log in, this seems to be a problem with your local machine configuration. EDDI appears to be functioning correctly.

Is there anything on your local system that might interrupt a connection or prevent EDDI from connecting to the Frontier API server at https://companion.orerve.net? Maybe you could try [troubleshooting your local firewall](https://technet.microsoft.com/en-us/library/cc766312%28v=ws.10%29.aspx?f=255&MSPPError=-2147217396)?

**Comment by Tkael** [comment]
I'm going to close this thread as not an EDDI issue, but I will continue to monitor it. If you think there's something more we can do to help troubleshoot this issue, I'll still see it here and try to do what I can to help.

**Comment by Razzledazzk** [comment]
Ok Tkael, thank you for the support.

I have looked into the firewall, but not been able to see if the error was there or not.

To start out I will do a reinstall of windows and the game only, and see if that fixes the issue on my end.
If that doenst fixes it, I am gonna call some tech support on the firewall and see if running without it fixes things - and if so, looking into what exceptions I need to tell the firewall to do to work.

Once again, thank you for your tremendeous work, and hopefully in a day or so, I can update in here and tell you it worked :) 👍

**Comment by Tkael** [comment]
Good luck!

**Comment by Razzledazzk** [comment]
Problem is now certainly on my end.

I disabled the firewall, completely went 100% no security, but still didnt work.
So I reinstalled windows and the game - still nothing.

Thought way out of the box and tried out the internet from my phone - yeah.. sarcasme inserted - and lo behold, API connects instantly.

After troubleshooting, then the firewall on the router isnt an issue - but through unknown reasons, then the switch I have to expand the network, have an issue with letting me connect to EDDI properly.

So now I am working towards a solution without that road block.

Thank you again for looking into the problem - in the future I will remember to try out a different network, if only briefly, as our cellphones gives us that option. I just never realised it could be there.

/Martin
